site/public/js/server.js

```diff
@@ -455,4 +455,77 @@ $(function() {
     }, 5000);
 
     setupCheckboxCells();
+});
+
+function setupNumericTextCells() {
+    $("input[class=option-small-box]").keydown(function(key){
+        var cell=this.id.split('-');
+        if(key.keyCode === 39){
+            if(this.selectionEnd == this.value.length){
+                $('#cell-'+cell[1]+'-'+(++cell[2])).focus();
+            }
+        }
+        else if(key.keyCode == 37){
+            if(this.selectionStart == 0){
+                $('#cell-'+cell[1]+'-'+(--cell[2])).focus();
+            }
+        }
+        else if(key.keyCode == 38){
+            $('#cell-'+(--cell[1])+'-'+cell[2]).focus();
+
+        }
+        else if(key.keyCode == 40){
+            $('#cell-'+(++cell[1])+'-'+cell[2]).focus();
+        }
+    });
+
+    $("input[class=option-small-box]").change(function() {
+        elem = this;
+        if(this.value == 0){
+            $(this).css("color", "#bbbbbb");
+        }
+        else{
+            $(this).css("color", "");
+        }
+        var scores = {};
+        var total = 0;
+        $(this).parent().parent().children("td.option-small-input, td.option-small-output").each(function() {
+            $(this).children(".option-small-box").each(function(){
+                if($(this).data('num') === true){
+                    total += parseFloat(this.value);
+                }
+                if($(this).data('total') === true){
+                    this.value = total;
+                }
+                else{
+                    scores[$(this).data("id")] = this.value;
+                }
+            });
+        });
+
+        submitAJAX(
+            buildUrl({'component': 'grading', 'page': 'simple', 'action': 'save_numeric'}),
+            {'csrf_token': csrfToken, 'user_id': $(this).parent().parent().data("user"), 'g_id': $(this).parent().parent().data('gradeable'), 'scores': scores},
+            function() {
+            },
+            function() {
+                $(elem).css("background-color", "#ff7777");
+            }
+        );
+    });
+}
+
+$(function() {
+    if (window.location.hash !== "") {
+        if ($(window.location.hash).offset().top > 0) {
+            var minus = 60;
+            $("html, body").animate({scrollTop: ($(window.location.hash).offset().top - minus)}, 800);
+        }
+    }
+
+    setTimeout(function() {
+        $('.inner-message').fadeOut();
+    }, 5000);
+
+    setupNumericTextCells();
```

Comment:
Just put `setupNumericTextCells()` in the above `$(function() {` block and get rid of this duplicate block.